src/components/TodoForm/TodoForm.tsx

```diff
@@ -0,0 +1,114 @@
+import { useState } from 'react';
+import './TodoForm.scss';
+import { User } from '../../types/user';
+import { Todo } from '../../types/todo';
+import { getNewTodoId } from '../../helpers/getNewPostId';
+import { getUser } from '../../helpers/getUser';
+
+interface Props {
+  todos: Todo[];
+  users: User[];
+  onSubmit: (newTodo: Todo) => void;
+}
+
+export const TodoForm: React.FC<Props> = ({ todos, users, onSubmit }) => {
+  const [title, setTitle] = useState('');
+  const [user, setUser] = useState('0');
+  const [submitting, setSubmitting] = useState(false);
+
+  const usersNames = users.map(({ name }) => name);
+
+  const isTitleError = title === '';
+  const isUserError = user === '0';
+
+  const handleTitleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const inputValue = e.target.value;
+    const pattern = /^[A-Za-zА-ЩЇІЄҐа-щїієґ0-9\s]+$/;
+
+    const cleanedInput = inputValue
+      .split('')
+      .filter(char => pattern.test(char) || char === ' ')
+      .join('');
```

Comment:
- With regex, you can test all value, you don't need to split, filter, etc
- Also, if you need to get rid of invalid characters you can use replace, it works with regex 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace